fenics_ice/solver.py

```diff
@@ -509,14 +533,27 @@ def solve_mom_eq(self, annotate_flag=None):
 
         newton_params = self.params.momsolve.newton_params
         picard_params = self.params.momsolve.picard_params
+        quad_degree = self.params.momsolve.quadrature_degree
         J_p = self.mom_Jac_p
 
-        momsolver = MomentumSolver(self.mom_F == 0,
+        dic_form = {"quadrature_degree": quad_degree}
+
+        if (quad_degree == -1):
+            momsolver = MomentumSolver(self.mom_F == 0,
                                    self.U,
                                    bcs=self.flow_bcs,
                                    J_p=J_p,
                                    picard_params=picard_params,
                                    solver_parameters=newton_params)
+        else:
+            momsolver = MomentumSolver(self.mom_F == 0,
+                                   self.U,
+                                   bcs=self.flow_bcs,
+                                   J_p=J_p,
+                                   picard_params=picard_params,
+                                   solver_parameters=newton_params,
+                                   form_compiler_parameters={"quadrature_degree": quad_degree})
```

Comment:
@jrmaddison  above you state: 

"I suspect it's better to use the symbolic version and then control the quadrature degree manually -- perhaps using dx(degree=quad_degree) to control the quadrature degree for a single term."

Is this statement actually directed at this? as in, should i not use the form_compiler_parameters kwarg?

Reply:
I mean it's better to have `EquationSolver` used in just one place, rather than in the two branches, as there are quite a lot of constructor arguments and we might want to change some of them. Having these in one place seems safer.